packages/typespec-test/eng/smoke-test.js

```diff
@@ -1,95 +1,136 @@
-      console.log("Run command:", cleanUpCmd);
-      execSync(cleanUpCmd, {
-        maxBuffer: MAX_BUFFER,
-      });
-    }
-    // Recovery the sources folder if we have
-    if (hasCustomization) {
-      const recoveryCmd = `mkdir -p ${join(
-        path,
-        "generated",
-        "typespec-ts",
-        "sources"
-      )}`;
-      console.log("Run command:", recoveryCmd);
-      execSync(recoveryCmd, {
-        maxBuffer: MAX_BUFFER,
-      });
+    if (await exists(join(path, "generated", "typespec-ts"))) {
+      const pathToRemove = join(path, "generated", "typespec-ts");
+      await rm(pathToRemove, { recursive: true, force: true });
     }
   } catch (e) {
     // do nothing
-    console.log("Preparation error:", e);
+    logger.log("Preparation error:", e);
   }
 
   try {
-    console.log("Run command:", generateCmd);
-    const result = execSync(generateCmd, {
-      maxBuffer: MAX_BUFFER,
-    });
-    console.log("Generated output:", result.toString("utf8"));
+    await npxCommand("tsp", tspArgs, workingDir, logger);
   } catch (e) {
-    console.error(Error(e.stdout.toString("utf8")));
-    process.exitCode = 1;
-  }
-}
-
-function copyFile(path) {
-  if (hasCustomizationFolder(path)) {
-    const customizationPath = join(
-      path,
-      "generated/typespec-ts/sources/generated/src"
-    );
-    const srcPath = join(path, "generated/typespec-ts");
-    const cp = `cp -rf ${customizationPath} ${srcPath}`;
-    console.log(cp);
-    const result = execSync(cp, {
-      maxBuffer: MAX_BUFFER,
-    });
-    console.log("Copy file output:", result.toString("utf8"));
+    logger.error(Error(e.toString("utf8")));
   }
 }
 
 const failed = [];
 
-function build(path) {
-  const command = `cd ${path}/generated/typespec-ts && npm install && npm run build`;
-  console.log(command);
+async function build(path, logger) {
+  const workingDir = join(path, "generated/typespec-ts");
   try {
-    const result = execSync(command, {
-      maxBuffer: MAX_BUFFER,
-    });
-    console.log("build output:", result.toString("utf8"));
+    await runCommand("npm", ["install"], workingDir, logger);
+    await runCommand("npm", ["run", "build"], workingDir, logger);
   } catch (e) {
     failed.push(path);
-    console.log(Error(e.stdout.toString("utf8")));
+    logger.error(Error(e.toString("utf8")));
     process.exitCode = 1;
   }
 }
 
-function hasCustomizationFolder(path) {
-  const customizationPath = join(path, "generated/typespec-ts/sources");
-  return existsSync(customizationPath);
+async function generateSmokeTest(path) {
+  const logger = createTaskLogger();
+  const contents = await readdir(path);
+  logger.log(`================Start ${path}===============`);
+  if (contents.includes("skip")) {
+    logger.log(`          ##### Skipped #####          `);
+  } else {
+    await generate(path, logger);
+    await build(path, logger);
+  }
+  logger.log(`================End ${path}===============`);
+  logger.flush();
```

Comment:
Will the all logging mix together with disordered messages?

Reply:
no they are now synchronized so all logs from the same task are going to together